File: crates/common/src/crate_overview.rs
use sea_orm::FromQueryResult;
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(
    Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize, FromQueryResult, ToSchema,
)]
pub struct CrateOverview {
    pub name: String,
    pub version: String,
    pub date: String,
    pub total_downloads: i64,
    pub description: Option<String>,
    pub documentation: Option<String>,
    pub is_cache: bool,
}

File: ui/src/types/crate_overview.ts
export type CrateOverview = {
    name: string
    version: string
    date: string
    total_downloads: number
    description?: string
    documentation?: string
    is_kellnr: boolean
    /** Alias for !is_kellnr - indicates if this is from crates.io cache */
    is_cache?: boolean
}
